<launch>

  <arg name="rviz" default="true" />
  <arg name="scanPeriod" default="0.1" />
  <arg name="confidenceThreshold" default="0.6" />
  <arg name="maskThreshold" default="0.6" />
  <arg name="matchSqDistThreshold" default="0.25" />

  <node pkg="loam_velodyne_semantic_mapping" type="multiScanRegistration" name="multiScanRegistration" output="screen">
    <param name="lidar" value="VLP-16" /> <!-- options: VLP-16  HDL-32  HDL-64E -->
    <param name="scanPeriod" value="$(arg scanPeriod)" />

    <remap from="/multi_scan_points" to="/velodyne_points"/>
  </node>

  <node pkg="loam_velodyne_semantic_mapping" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne_semantic_mapping" type="laserMapping" name="laserMapping" output="screen">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne_semantic_mapping" type="transformMaintenance" name="transformMaintenance" output="screen">
  </node>

  <node pkg="loam_velodyne_semantic_mapping" type="semanticMapping" name="semanticMapping" output="screen">
      <param name="confidenceThreshold" value="$(arg confidenceThreshold)" />
      <param name="maskThreshold" value="$(arg maskThreshold)" />
      <param name="matchSqDistThreshold" value="$(arg matchSqDistThreshold)" />
  </node>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_velodyne_semantic_mapping)/rviz_cfg/loam_velodyne_semantic_mapping.rviz" />
  </group>

</launch>
